File: software/examples/ruby/example_configuration.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/brick_stepper'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = '9eggGYnDtnd' # Change to your UID

ipcon = IPConnection.new # Create IP connection
stepper = BrickStepper.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

stepper.set_motor_current 800 # 800mA
stepper.set_step_mode 8 # 1/8 step mode
stepper.set_max_velocity 2000 # Velocity 2000 steps/s

# Slow acceleration (500 steps/s^2), 
# Fast deacceleration (5000 steps/s^2)
stepper.set_speed_ramping 500, 5000 

stepper.enable
stepper.set_steps 60000 # Drive 60000 steps forward

puts 'Press key to exit'
$stdin.gets

File: software/examples/go/example_configuration.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/stepper_brick"
)

const ADDR string = "localhost:4223"
const UID string = "XXYYZZ" // Change XXYYZZ to the UID of your Stepper Brick.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	stepper, _ := stepper_brick.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	stepper.SetMotorCurrent(800) // 800 mA
	stepper.SetStepMode(8)       // 1/8 step mode
	stepper.SetMaxVelocity(2000) // Velocity 2000 steps/s

	// Slow acceleration (500 steps/s^2),
	// Fast deacceleration (5000 steps/s^2)
	stepper.SetSpeedRamping(500, 5000)

	stepper.Enable()        // Enable motor power
	stepper.SetSteps(60000) // Drive 60000 steps forward

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

	// Stop motor before disabling motor power
	stepper.Stop() // Request motor stop
	stepper.SetSpeedRamping(500,
		5000) // Fast deacceleration (5000 steps/s^2) for stopping
	time.Sleep(400 * time.Millisecond) // Wait for motor to actually stop: max velocity (2000 steps/s) / decceleration (5000 steps/s^2) = 0.4 s
	stepper.Disable()                  // Disable motor power
}
